packages/api/src/external/carequality/command/cq-directory/get-cq-directory-entries.ts

```diff
@@ -0,0 +1,69 @@
+import { QueryTypes } from "sequelize";
+import { Pagination, sortForPagination } from "../../../../command/pagination";
+import { CQDirectoryEntry2 } from "../../cq-directory";
+import { HIEDirectoryEntryViewModel } from "../../models/hie-directory-view";
+
+export async function getHieDirectoryEntriesByFilter(
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  { filter, pagination }: { filter: string; pagination: Pagination }
+): Promise<CQDirectoryEntry2[]> {
+  const sequelize = HIEDirectoryEntryViewModel.sequelize;
+  if (!sequelize) throw new Error("Sequelize not found");
+
+  const querySelect = `SELECT * FROM ${HIEDirectoryEntryViewModel.tableName} WHERE 1 = 1 `;
+
+  const queryFTS =
+    querySelect +
+    (filter
+      ? ` AND (search_criteria @@ websearch_to_tsquery('english', :filter) OR id = :filter)`
+      : "");
+
+  const { toItem, fromItem } = pagination ?? {};
+  const toItemStr = toItem ? ` AND id >= :toItem` : "";
+  const fromItemStr = fromItem ? ` AND id <= :fromItem` : "";
+  const queryPagination = queryFTS + " " + [toItemStr, fromItemStr].filter(Boolean).join("");
+
+  const queryOrder = queryPagination + " ORDER BY id " + (toItem ? "ASC" : "DESC");
+
+  const { count } = pagination ?? {};
+  const queryFinal = queryOrder + (count ? ` LIMIT :count` : "");
```

Comment:
This is the same logic as on get-patient just wondering if there is some refactoring we could do.

Reply:
yee good call, ill pull it out to something

Reply:
addressed